crypto/ex_data.c

```diff
@@ -287,8 +287,6 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         CRYPTOerr(CRYPTO_F_CRYPTO_DUP_EX_DATA, ERR_R_MALLOC_FAILURE);
         return 0;
     }
-    if (!CRYPTO_set_ex_data(to, mx - 1, NULL))
-        goto err;
```

Comment:
Here's the key thing:

'If the pointer contained in B<*pptr> is not modified by the dup_func(), then both B<to> and B<from> will point to the same data.'

So, you can update the `*pptr` to be the original data (which can be retrieved via `CRYPTO_get_ex_data(to, ...)`)

If you don't otherwise manage the pointers, things get lost, and hairy.